src/uploadNotionImagesToCloudinary.ts

```diff
@@ -66,11 +81,13 @@ export default async function uploadNotionImagesToCloudinary({
       const coverImage = await downloadImageToBase64(coverUrl);
       log.debug("Image downloaded");
 
+      const filenameFromTitle = title ? makeFilenameFromCaption(title, 200) : undefined
+
       const { url: coverExternalUrl } = await cloudinaryClient.uploadImage(
         `data:image/jpeg;base64,${coverImage}`,
         {
           folder: `${cloudinaryUploadFolder}/${page.id}`,
-          // @todo: add filename here, pulling from the Notion API page title
+          public_id: filenameFromTitle,
```

Comment:
I did not add any id to the name, because the title should be unique here.

Even if someone uses the same title as caption in the body => body images get the imageBlock suffix'ed

Reply:
BUT this can only be a problem if you upload a new cover image to notion, run the script, and then the user still sees the old image (without any extra CDN invalidaton).
Maybe we should add a little hash (nanoid, random number, ..., whatever)? 

The same actually applies to the body images, when you just "replace" the image and upload a new one. If caption stays the same, the filename would stay the same. And same caching issue.

Reply:
I didn't think this would be a common situation when making this lib, but to make it more robust, adding a little hash at the end of the filename would work. As a workaround, I imagine from notion one can edit the image URL and add a query param like `?v=1` to get a fresh copy.